x-pack/plugins/cloud_integrations/cloud_chat/public/plugin.tsx

```diff
@@ -31,30 +38,47 @@ interface CloudChatConfig {
   trialBuffer: number;
 }
 
-export class CloudChatPlugin implements Plugin {
+export interface CloudChatPluginStart {
+  Chat: React.ComponentType;
+}
+
+export class CloudChatPlugin
+  implements Plugin<void, CloudChatPluginStart, CloudChatSetupDeps, CloudChatStartDeps>
+{
   private readonly config: CloudChatConfig;
   private chatConfig$ = new ReplaySubject<ChatConfig>(1);
+  private Chat: React.ComponentType | undefined;
 
   constructor(initializerContext: PluginInitializerContext<CloudChatConfig>) {
     this.config = initializerContext.config.get();
   }
 
-  public setup(core: CoreSetup, { cloud, security }: CloudChatSetupDeps) {
-    this.setupChat({ http: core.http, cloud, security }).catch((e) =>
+  public setup(core: CoreSetup, { cloud, security, cloudChatProvider }: CloudChatSetupDeps) {
+    this.setupChat({ http: core.http, cloud, security, cloudChatProvider }).catch((e) =>
       // eslint-disable-next-line no-console
       console.debug(`Error setting up Chat: ${e.toString()}`)
     );
-
     const CloudChatContextProvider: FC = ({ children }) => {
       // There's a risk that the request for chat config will take too much time to complete, and the provider
       // will maintain a stale value.  To avoid this, we'll use an Observable.
       const chatConfig = useObservable(this.chatConfig$, undefined);
       return <ServicesProvider chat={chatConfig}>{children}</ServicesProvider>;
     };
     cloud.registerCloudService(CloudChatContextProvider);
+    cloudChatProvider.registerChatProvider(() => this.Chat);
```

Comment:
ultra-nit: Is `CloudChatProvider` actually a `CloudChatService` that serves a downstream provider?